uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequestStatusMatrix.php

```diff
@@ -151,9 +151,7 @@ private function getStatusMatrixForCurrentUser($leaveRequestContactID) {
 
     if ($currentUserID == $leaveRequestContactID) {
       $statusMatrix = $this->getStaffStatusMatrix();
-    }
-
-    if ($this->shouldUseManagerMatrixForCurrentUser($leaveRequestContactID)) {
+    } elseif ($this->shouldUseManagerMatrixForCurrentUser($leaveRequestContactID)) {
```

Comment:
The Admin when requesting for leave is a staff member as well as an admin. Using the elseif makes sure that once the logic detects that you are requesting leave for yourself, use the status matrix irrespective of any role you may have. This helps eliminate unnecessary checks in `shouldUseManagerMatrixForCurrentUser` because it also runs some queries to determine  if the user is a leave approver or admin. More like a short circuit.